app/src/main/java/com/github/sdp_begreen/begreen/fragments/SendPostFragment.kt

```diff
@@ -222,41 +221,46 @@ class SendPostFragment : Fragment() {
 
     private suspend fun contestsUpdateScores(
         trashCategory: TrashCategory,
-        location: CustomLatLng,
+        location: CustomLatLng
     ) {
+        val userId = connectedUserViewModel.currentUser.value?.id
+        val participationMap =
+            eventsFragmentViewModel.participationMap.dropWhile { it.isEmpty() }.first()
+        val contests = eventsFragmentViewModel.allEvents.dropWhile { it.isEmpty() }.first()
 
-        connectedUserViewModel.currentUser.value?.id?.also { participantId ->
-            val participationMap =
-                eventsFragmentViewModel.participationMap.dropWhile { it.isEmpty() }.first()
-
-            eventsFragmentViewModel.allEvents.dropWhile { it.isEmpty() }.first()
-                .forEach { contest ->
-
-                    // Check that the location is valid before checking if the user is in range
-                    location.toMapLocation()?.also {
-                        if (participationMap[contest.id] == true &&
-                            contest.isActive() == true &&
-                            contest.isInRange(it) == true
-                        ) {
-                            val updatedParticipant = eventParticipantService.getParticipant(
-                                RootPath.CONTESTS,
-                                contest.id!!,
-                                participantId,
-                                ContestParticipant::class.java
-                            )
-                            eventParticipantService.addParticipant(
-                                RootPath.CONTESTS,
-                                contest.id!!,
-                                updatedParticipant.copy(
-                                    score = updatedParticipant.score?.plus(trashCategory.value)
-                                        ?: trashCategory.value
-                                )
-                            )
-                        }
-                    }
-                }
+        contests.forEach { contest ->
+            processContest(contest, userId, participationMap, trashCategory, location)
         }
+    }
 
+    private suspend fun processContest(
+        contest: Contest,
+        userId: String?,
+        participationMap: Map<String, Boolean>,
+        trashCategory: TrashCategory,
+        location: CustomLatLng
+    ) {
+        location.toMapLocation()?.let {
+            if (participationMap[contest.id] == true &&
+                contest.isActive() == true &&
+                contest.isInRange(it) == true
+            ) {
+                val updatedParticipant = eventParticipantService.getParticipant(
+                    RootPath.CONTESTS,
+                    contest.id!!,
```

Comment:
It appears that you are currently enforcing the usage of `userId!!`, but it seems that there are no check in place to verify its non-null state. Instead, it might be more beneficial to declare `userId` as a non-null String in the method parameter, requiring the caller of the method to ensure its non-null status before invoking this helper method. Although I didn't mention this earlier when all the code was in the same block, now that you've separated the behavior into a separate method, it would be prudent to safeguard against incorrect call of this method to avoid any crash.

Reply:
good catch!